File: 15312.py
strCnt = [3, 2, 1, 2, 3, 3, 2, 3, 3, 2, 2, 1, 2, 2, 1, 2, 2, 2, 1, 2, 1, 1, 1, 2, 2, 1]

s1 = input()
s2 = input()

love = []
for i in range(len(s1)):
    love.append(strCnt[ord(s1[i]) - ord('A')])
    love.append(strCnt[ord(s2[i]) - ord('A')])

while len(love) > 2:
    for i in range(len(love) - 1):
        love[i] = (love[i] + love[i + 1]) % 10
    love = love[:len(love)-1]

print(love[0], love[1], sep='')

File: 15312.cpp
#include <iostream>
#include <queue>
using namespace std;

int strCnt[26] = { 3, 2, 1, 2, 3, 3, 2, 3, 3, 2, 2, 1, 2, 2, 1, 2, 2, 2, 1, 2, 1, 1, 1, 2, 2, 1 };

int main() {
	string s1, s2; cin >> s1 >> s2;

	queue<int> love;
	for (int i = 0; i < s1.length(); i++) {
		love.push(strCnt[s1[i] - 'A']);
		love.push(strCnt[s2[i] - 'A']);
	}

	while (love.size() > 2) {
		int len = love.size();
		for (int i = 0; i < len - 1; i++) {
			int a1 = love.front();
			love.pop();
			love.push((a1 + love.front()) % 10);
		}
		love.pop();
	}

	cout << love.front() << love.back() << endl;
}
